### DISCUSSION TITLE
Line Spacing in Datagrid

### DISCUSSION THREAD

**Discussion by stephannn** [AvaloniaUI/Avalonia#17615] [Q&A]
Hello,

I would like to ask if it is possible to reduce the space below and above text in a datagrid. For example, in a WPF you can see that the spaces between the rows almost do not exist.
WPF:
![image](https://github.com/user-attachments/assets/974a1a93-00f0-4415-9226-877860762e9c)

On the other hand, in Avalonia it has more space in between
![image](https://github.com/user-attachments/assets/f26e8a66-c09f-40ca-878a-4600ed68b6ac)

My Datagrid looks like that:
```
                    <DataGrid x:Name="datagrid_information" ItemsSource="{Binding Informationdgm, Mode=OneWay, UpdateSourceTrigger=PropertyChanged}" 
                        IsReadOnly="True" AutoGenerateColumns="False" HeadersVisibility="None" CanUserReorderColumns="False" CanUserResizeColumns="False" CanUserSortColumns="False" BorderThickness="0.1" >
                            <DataGrid.Columns>
                                <DataGridTextColumn FontSize="12" Binding="{Binding Attribute}" Header="Attribute" Width="Auto"/>
                                <DataGridTextColumn FontSize="12" Binding="{Binding Value}" Header="Value" Width="Auto"/>
                            </DataGrid.Columns>
                    </DataGrid>
```
Thanks
Stephan

**Comment by pyking-wang** [comment]
```
<DataGrid.Styles>
  <Style Selector="DataGridRow">
	<Setter Property="Height" Value="20"/>
  </Style>
</DataGrid.Styles>
```

**Reply by stephannn** [reply]
Hi,
sorry, I forgot to mention that some rows have two lines:
![image](https://github.com/user-attachments/assets/74eea1be-cd3a-455e-bbf2-c0e5a95c91b7)
By setting the value that way, all rows are shrinked

**Reply by pyking-wang** [reply]
Perhaps you can try setting the MaxHeight or MinHeight property?

**Reply by stephannn** [reply]
I think the issue is that _MinHeight_, when setting it to 20 will be ignored because the font size is bigger with the spaces above. Same with _MaxHeight_ vice versa

**Comment by stevemonaco** [comment]
Fluent theme isn't intended to be compact -- it's sparse to make touch easier. That said, I think you need to write a style selector for the following part of the theme:

https://github.com/AvaloniaUI/Avalonia/blob/bed48a7bb62a2deb4e246065a093d36955b28c57/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L84-L90

Such as:

```xml
<Style Selector="DataGridCell">
    <Setter Property="MinHeight" Value="20" />
</Style>
```

I don't see any other suspicious margins, paddings, or explicit sizes from a quick look.
